app/src/app/[locale]/serverAction/page.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import { useFormState } from "react-dom";
+
+import {
+  Button,
+  Grid,
+  GridContainer,
+  Label,
+  TextInput,
+} from "@trussworks/react-uswds";
+
+import { updateServerData } from "../../serverActions/serverActionExample";
+
+const initialFormState = {
+  name: "",
+  email: "",
+};
+
+export default function SimpleForm() {
+  const [formData, updateFormData] = useFormState(
+    updateServerData,
+    initialFormState
+  );
+
+  const hasReturnedFormData = formData.name || formData.email;
+
+  return (
+    <GridContainer>
+      <Grid row>
+        <Grid col={12}>
+          <div className="usa-card">
+            <div className="usa-card__header">
+              <h2>Server Action Example</h2>
+            </div>
+            <div className="usa-card__body">
+              <form action={updateFormData}>
+                <Label htmlFor="name">Name</Label>
+                <TextInput
+                  id="name"
+                  name="name"
+                  type="text"
+                  defaultValue={formData.name}
+                />
+
+                <Label htmlFor="email">Email</Label>
+                <TextInput
+                  id="email"
+                  name="email"
+                  type="email"
+                  defaultValue={formData.email}
+                  className="margin-bottom-1"
+                />
+
+                <Button type="submit">Submit</Button>
```

Comment:
I think it'd be helpful to demonstrate how to show a "pending" button state, where the button is disabled to prevent duplicate submissions, using [`useFormStatus`](https://react.dev/reference/react-dom/hooks/useFormStatus).

Reply:
`useFormStatus` has to be nested in the form so broke it out to a separate component